java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -0,0 +1,1585 @@
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector;
+
+import static org.apache.arrow.memory.util.LargeMemoryUtil.capAtMaxInt;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.arrow.memory.ArrowBuf;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.OutOfMemoryException;
+import org.apache.arrow.memory.util.ArrowBufPointer;
+import org.apache.arrow.memory.util.ByteFunctionHelpers;
+import org.apache.arrow.memory.util.CommonUtil;
+import org.apache.arrow.memory.util.hash.ArrowBufHasher;
+import org.apache.arrow.util.Preconditions;
+import org.apache.arrow.vector.compare.VectorVisitor;
+import org.apache.arrow.vector.ipc.message.ArrowFieldNode;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.util.CallBack;
+import org.apache.arrow.vector.util.OversizedAllocationException;
+import org.apache.arrow.vector.util.TransferPair;
+
+/**
+ * BaseVariableWidthViewVector is a base class providing functionality for strings/bytes types in view format.
+ *
+ */
+public abstract class BaseVariableWidthViewVector extends AbstractVariableWidthVector {
+  private static final int DEFAULT_RECORD_BYTE_COUNT = 16;
+  private static final int INITIAL_BYTE_COUNT = INITIAL_VALUE_ALLOCATION * DEFAULT_RECORD_BYTE_COUNT;
+  private static final int MAX_BUFFER_SIZE = (int) Math.min(MAX_ALLOCATION_SIZE, Integer.MAX_VALUE);
+  private int lastValueCapacity;
+  private long lastValueAllocationSizeInBytes;
+
+  public static final int OFFSET_WIDTH = 4; /* 4 byte unsigned int to track offsets */
+  /* protected members */
+  protected static final int INLINE_SIZE = 12; /* 12 byte unsigned int to track inline views*/
+  protected static final int VIEW_BUFFER_SIZE = 16; /* 16 byte default size for each view*/
+  protected static final int LENGTH_WIDTH = 4; /* the first 4 bytes of view are allocated for length*/
+  protected static final int PREFIX_WIDTH = 4; /* the second 4 bytes of view are allocated for prefix width*/
+  protected static final int BUF_INDEX_WIDTH = 4; /* third 4 bytes of view are allocated for buffer index*/
+  protected static final byte[] emptyByteArray = new byte[]{};
+  protected ArrowBuf validityBuffer;
+  protected ArrowBuf valueBuffer;
+  protected ArrowBuf offsetBuffer;
+  protected int valueCount;
+  protected int lastSet;
+  protected final Field field;
+  protected List<ArrowBuf> dataBuffers;
+
+  /**
+   * Constructs a new instance.
+   *
+   * @param field The field materialized by this vector
+   * @param allocator The allocator to use for creating/resizing buffers
+   */
+  public BaseVariableWidthViewVector(Field field, final BufferAllocator allocator) {
+    super(allocator);
+    this.field = field;
+    lastValueAllocationSizeInBytes = INITIAL_BYTE_COUNT;
+    // -1 because we require one extra slot for the offset array.
```

Comment:
This isn't true...We don't have offsets...

Reply:
Related question: 

Since offset is not a part of this. The number of elements the ViewVarcharVector can hold vs number of elements VarCharVector can hold should be different right?

Reply:
I mean yes? One is view_buffer / 16 and the other is offsets_buffer / offset_width - 1

Reply:
But I'm trying to say you should _not_ be referencing the VarCharVector as the base of the implementation here

Reply:
Yes indeed.